website/src/components/genspectrum/GsLineageFilter.tsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useRef } from 'react';
+
+import '@genspectrum/dashboard-components';
+
+export function GsLineageFilter<Lineage extends string>({
+    lapisField,
+    initialValue,
+    placeholderText,
+    width,
+    onLineageChange = () => {},
+}: {
+    lapisField: Lineage;
+    initialValue?: string;
+    placeholderText?: string;
+    width?: string;
+    onLineageChange?: (location: { [key in Lineage]: string | undefined }) => void;
+}) {
+    const lineageFilterRef = useRef<HTMLElement>();
+
+    useEffect(() => {
+        const handleLineageChange = (event: CustomEvent) => {
+            onLineageChange(event.detail);
+        };
+
+        if (lineageFilterRef && lineageFilterRef.current) {
+            lineageFilterRef.current.addEventListener('gs-lineage-filter-changed', handleLineageChange);
+        }
+
+        return () => {
+            if (lineageFilterRef && lineageFilterRef.current) {
```

Comment:
```suggestion
        if (lineageFilterRef.current) {
            lineageFilterRef.current.addEventListener('gs-lineage-filter-changed', handleLineageChange);
        }

        return () => {
            if (lineageFilterRef.current) {
```

AFAIK the ref object itself is always defined?